doc/api/http2.md

```diff
@@ -0,0 +1,727 @@
+# HTTP2
+
+> Stability: 1 - Experimental
+
+The `http2` module provides an implementation of the [HTTP/2][] protocol. It
+can be accessed using:
+
+```js
+const http2 = require('http2');
+```
```

Comment:
If it's behind a flag (thanks for the reminder btw) then I don't believe that we'll need to do anything different here. We'll just need to document the flag for the time being. I'll work on getting the flag added this week